packages/common/src/dto/availability-information.dto.ts

```diff
@@ -1,13 +1,20 @@
-import { IsBoolean, IsIn, IsNumber, Length, ValidateNested } from 'class-validator';
+import { IsArray, IsBoolean, IsIn, IsNumber, ValidateNested } from 'class-validator';
 
 import { DeploymentDurations, HealthAuthorities } from '../interfaces';
 import { PlacementPreferencesDTO } from '.';
 
 export class AvailabilityDTO {
+  constructor(base: AvailabilityDTO) {
```

Comment:
we can use 'plainToClass' method from 'class-transformer' for this for assignment and skip having a constructor.  Use `Object.assign`